### PR TITLE
Set up AppVeyor to build Windows wheels

### LINKED ISSUE #149: Skip build option
Currently, only certain exceptions are caught before trying to build without speedups. Something in my environment causes `ValueError` to be raised, which means I can't install logbook.

Perhaps there should be a way to force skip of the compile step, or `ValueError` can be checked and raise `BuildFailed`.

```
Collecting logbook
  Using cached Logbook-0.10.0.tar.gz
Requirement already satisfied (use --upgrade to upgrade): six>=1.4.0 in c:\python34\lib\site-packages (from logbook)
Installing collected packages: logbook
  Running setup.py install for logbook
    Complete output from command c:\Python34\python.exe -c "import setuptools, tokenize;__file__='C:\\Users\\FRAZER~1\\AppData\\Local\\Temp\\pip-build-kco3r3mt\\logbook\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record C:\Users\FRAZER~1\AppData\Local\Temp\pip-ymatx5nb-record\install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.4
    creating build\lib.win-amd64-3.4\logbook
    copying logbook\base.py -> build\lib.win-amd64-3.4\logbook
    copying logbook\compat.py -> build\lib.win-amd64-3.4\logbook
    copying logbook\concurrency.py -> build\lib.win-amd64-3.4\logbook
    copying logbook\handlers.py -> build\lib.win-amd64-3.4\logbook
    copying logbook\helpers.py -> build\lib.win-amd64-3.4\logbook
    copying logbook\more.py -> build\lib.win-amd64-3.4\logbook
    copying logbook\notifiers.py -> build\lib.win-amd64-3.4\logbook
    copying logbook\queues.py -> build\lib.win-amd64-3.4\logbook
    copying logbook\ticketing.py -> build\lib.win-amd64-3.4\logbook
    copying logbook\_fallback.py -> build\lib.win-amd64-3.4\logbook
    copying logbook\_termcolors.py -> build\lib.win-amd64-3.4\logbook
    copying logbook\__init__.py -> build\lib.win-amd64-3.4\logbook
    running build_ext
    building 'logbook._speedups' extension
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\FRAZER~1\AppData\Local\Temp\pip-build-kco3r3mt\logbook\setup.py", line 133, in <module>
        run_setup(True)
      File "C:\Users\FRAZER~1\AppData\Local\Temp\pip-build-kco3r3mt\logbook\setup.py", line 124, in run_setup
        **extra
      File "c:\Python34\lib\distutils\core.py", line 148, in setup
        dist.run_commands()
      File "c:\Python34\lib\distutils\dist.py", line 955, in run_commands
        self.run_command(cmd)
      File "c:\Python34\lib\distutils\dist.py", line 974, in run_command
        cmd_obj.run()
      File "c:\Python34\lib\site-packages\setuptools\command\install.py", line 61, in run
        return orig.install.run(self)
      File "c:\Python34\lib\distutils\command\install.py", line 539, in run
        self.run_command('build')
      File "c:\Python34\lib\distutils\cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "c:\Python34\lib\distutils\dist.py", line 974, in run_command
        cmd_obj.run()
      File "c:\Python34\lib\distutils\command\build.py", line 126, in run
        self.run_command(cmd_name)
      File "c:\Python34\lib\distutils\cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "c:\Python34\lib\distutils\dist.py", line 974, in run_command
        cmd_obj.run()
      File "C:\Users\FRAZER~1\AppData\Local\Temp\pip-build-kco3r3mt\logbook\setup.py", line 83, in run
        build_ext.run(self)
      File "c:\Python34\lib\distutils\command\build_ext.py", line 339, in run
        self.build_extensions()
      File "c:\Python34\lib\distutils\command\build_ext.py", line 448, in build_extensions
        self.build_extension(ext)
      File "C:\Users\FRAZER~1\AppData\Local\Temp\pip-build-kco3r3mt\logbook\setup.py", line 89, in build_extension
        build_ext.build_extension(self, ext)
      File "c:\Python34\lib\distutils\command\build_ext.py", line 503, in build_extension
        depends=ext.depends)
      File "c:\Python34\lib\distutils\msvc9compiler.py", line 460, in compile
        self.initialize()
      File "c:\Python34\lib\distutils\msvc9compiler.py", line 371, in initialize
        vc_env = query_vcvarsall(VERSION, plat_spec)
      File "c:\Python34\lib\site-packages\setuptools\msvc9_support.py", line 52, in query_vcvarsall
        return unpatched['query_vcvarsall'](version, *args, **kwargs)
      File "c:\Python34\lib\distutils\msvc9compiler.py", line 287, in query_vcvarsall
        raise ValueError(str(list(result.keys())))
    ValueError: ['path']
```

### PR DESCRIPTION
**Pull request by RazerM**
Goal: fix tests that fail on Windows (#119), build wheels for Windows (also fixes #149).

I've added GitHub deployment, so that tagging a commit uploads the wheel files to a draft GitHub release for that tag. You'll need to generate your own [GitHub token](https://github.com/settings/tokens), and then encrypt it on the AppVeyor website using the **Account > Encrypt data** tool for the following part of appveyor.yml:

```
  auth_token:
    secure: <encrypted key>
```

I'm not sure if you want to set up some alternative that helps you download the files for uploading to PyPI.

### CODE REVIEW

File: logbook/more.py [hunk 1]
```diff
@@ -268,7 +268,7 @@ def __init__(self, arguments, stdin_format=None,
         self._subprocess = subprocess
 
     def emit(self, record):
-        args = [arg.format(record=record).encode(self.encoding)
+        args = [arg.format(record=record)
```
Comment (RazerM) [inline logbook/more.py]: I'm not sure why the parameters were being encoded before, I can't see in the subprocess documentation that bytes are required.

File: tests/test_queues.py [hunk 2]
```diff
@@ -71,6 +72,8 @@ def __call__(self):
 
 @require_module('multiprocessing')
 def test_multi_processing_handler():
+    if os.getenv('APPVEYOR') == 'True':
```
Comment (RazerM) [inline tests/test_queues.py]: These tests passed on my local Windows install, but failed for some (not all) Python versions on AppVeyor. Perhaps I should only skip on the specific Python versions that fail now that I think of it...

**Comment by vmalloc** [comment]
I think we'll pass on deployment via Appveyor. If at all, we will use Travis for that, but we're passing on it right now because of the script for pushing a release that also pushes the docs, rewrites the changelog etc.

**Comment by vmalloc** [comment]
Oh sorry - I missed the fact that it's deploying wheels, not the pypi release. How exactly does this work? Do I encrypt the token with Appveyor first?

**Comment by RazerM** [comment]
I have AppVeyor running tests with and without cython. When cython is built, a wheel is built. [See here].(https://ci.appveyor.com/project/razerm/logbook/build/job/98yaa1fh59qd5fh6/artifacts)

AppVeyor has a number of ways to send the wheel files elsewhere, but all that matters is that whoever uploads to PyPI gets the wheels to also upload those to PyPI.

For the GitHub token, encrypt it with the AppVeyoe tool then place it in the yml.

**Comment by vmalloc** [comment]
Thanks. I haven't gotten around to setting the GH key yet - hopefully later this week.

**Pull request merged**